brokers/vscode/broker.go

```diff
@@ -112,14 +121,22 @@ func (b *Broker) processPlugin(meta model.PluginMeta) error {
 	unpackedPath := filepath.Join(workDir, "plugin")
 
 	// Download an archive
-	b.PrintDebug("Downloading archive '%s' for plugin '%s:%s' to '%s'", url, meta.ID, meta.Version, archivePath)
-	err = b.download(url, archivePath, meta)
-	if err != nil {
-		return err
+	if url != "" {
+		b.PrintDebug("Downloading archive '%s' for plugin '%s:%s' to '%s'", url, meta.ID, meta.Version, archivePath)
+		err = b.ioUtil.Download(url, archivePath)
+		if err != nil {
```

Comment:
Doesn't seem like this error case is covered with tests. And a successful case is not covered too.

Reply:
* TestProcessPluginWithUrl - a successful case

Not sure what should we expect from testing a failure.

Reply:
I mean tests:
- check the error of downloading binaries from URL
- check verifies that download call was executed

Reply:
`TestProcessPluginBrokenUrl`